Misc. Updates

blakeneyops opened the pull request:
Updates to support integration into other contexts as well as standardization. This would be targeting a 0.0.6 alpha release.

yuhunglin commented on certs/main.tf:
```diff
@@ -148,9 +213,9 @@ module "cluster" {
 
   # LC parameters
-  ami              = "${var.ami}"
+  ami              = "${coalesce(lookup(var.ami_region_lookup, var.region), var.ami_custom)}"
```
👍  ah nice, reflecting the lookup changes from the other cert gen module.

yuhunglin commented on certs/variables.tf:
```diff
@@ -74,8 +75,14 @@ variable "s3_bucket_prefix" {
 }
 
-variable "cidr_whitelist" {
+variable "vpn_whitelist" {
```
oh, a split whitelist? what's the benefit?

blakeneyops replied:
Just additional granularity if desired, as the SSH should really be an admin backdoor only.

yuhunglin replied:
👌

yuhunglin commented on generate-certs/main.tf:
```diff
@@ -37,8 +36,8 @@ resource "aws_iam_role_policy" "s3_gen_certs" {
         "s3:PutObject"
       ],
       "Resource": [
-        "arn:aws:s3:::${replace(var.s3_root_path,"/(\/)+$/","")}",
```
ah, this fun regex helped take care of cases where the root path might have trailing slashes.

blakeneyops replied:
Strangest thing, I cannot get Terraform to allow this now. Getting the following:

```
Error loading Terraform: Error downloading modules: module openvpn_server: Error parsing .terraform/modules/bf21773e112cafd6226a9e0ab47e5ce3/main.tf: At 193:59: illegal char escape
```

cristobal23 replied:
same. @blakeneyops what did you do to fix it?

yuhunglin commented on generate-certs/main.tf:
```diff
@@ -47,17 +46,17 @@ resource "aws_iam_role_policy" "s3_gen_certs" {
         "s3:List*"
       ],
       "Resource": [
-        "arn:aws:s3:::${element(split("/", var.s3_root_path), 0)}"
```
root path was also used to try and anticipate cases where the certs wasn't going to be mounted at the base of an s3 bucket.

yuhunglin commented on generate-certs/main.tf:
```diff
@@ -79,63 +78,80 @@ EOF
-    openvpn_host      = "${var.openvpn_host}"
-    force_cert_regen  = "${var.force_cert_regen}"
-    s3_push_dryrun    = "${var.s3_push_dryrun}"
+    active_clients   = "${var.active_clients}"
```
👍  🔤

yuhunglin commented on generate-certs/templates/user_data.tpl:
```diff
@@ -2,20 +2,20 @@
 manage_etc_hosts: True
 
 runcmd:
-  - echo "S3_REGION=\"ap-northeast-1\"" > /etc/default/openvpn-cert-generator
```
ah yes, thanks

blakeneyops commented:
Fixed. Turned out the only place it was complaining about the regex was as a variable input to the template. I moved the regex into the template and that resolved it.

yuhunglin commented:
👌
